app/controllers/album.js

```diff
@@ -12,8 +14,37 @@ exports.list = (req, res, next) => {
     })
     .catch(error => {
       logger.error(error);
-      return res.status(502).send(error);
+      return res.status(500).send(error);
     });
 };
 
-exports.buy = (req, res, next) => {};
+exports.buy = (req, res, next) => {
+  logger.info('Starting buy album');
+  const tokenString = req.headers.authorization.replace('Bearer ', '');
```

Comment:
Maybe you should do this in the middleware; when you check that the user token is valid, you can place the decoded version of it inside the `request` object, and simply access it here